lib/active_merchant/billing/gateways/xpay.rb

```diff
@@ -253,16 +248,21 @@ def build_request_url(action, id = nil)
         base_url + endpoint
       end
 
-      def success_from(response)
-        SUCCESS_MESSAGES.include?(response.dig('operation', 'operationResult'))
+      def success_from(action, response)
```

Comment:
💬 Comment/Style =>

Leaving a comment here for the `build_request_url` method, a simpler implementation:

```ruby
      def build_request_url(action, id = nil)
        "#{test? ? test_url : live_url}#{ENDPOINTS_MAPPING[action.to_sym] % id}"
      end
```